dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentVersionResource.java

```diff
@@ -111,6 +113,8 @@ public Response findVersions(@Context final HttpServletRequest request,
         try {
 
            final Identifier identifierObj = getIdentifier(identifier, user);
+           // this is included in case the the query string ?render=true is included
+           ThreadContextUtil.getOrCreateContext().addContextMap(new HttpRequestMapWrapper(request));
```

Comment:
Don't we already have the request on a Threadlocal?  I am not sure we need all this code.

Reply:
Yes but I do not want to couple the Request with the Transformer or any other biz logic, so I wrapped it into the map and put it into the thread context to be use for whatever needs it without using directly the request.
One of the advantages is that the ContextMap could be supplied/implemented by something else rather than a request to provide the information and the code continue working, for instance in a separated thread, job, etc.